apps/web/src/hooks/useSessionRedirect.ts

```diff
@@ -38,10 +38,19 @@ export const useSessionRedirect = (shouldRedirect = true) => {
       } else {
         redirectToLogin(router)
       }
-    } else if (workspaces.data.length === 0) {
-      signOut()
     } else {
-      router.replace(`/workspaces/${workspaces.data[0].id}/documents`)
+      const user = session.data
+      const workspace =
+        workspaces.data.find(
+          (workspace) => workspace.id === user.lastVisitedWorkspaceId
+        ) ??
+        workspaces.data.find((workspace) => workspace.ownerId === user.id) ??
+        workspaces.data[0]
+      if (workspace) {
+        router.replace(`/workspaces/${workspaces.data[0].id}/documents`)
```

Comment:
Can't `workspaces.data[0]` be undefined, technically speaking? Shouldn't we have kept that `length` check above?

Reply:
This is a bug, it should use `workspace.id`